tests/Settings/TestSettingFunction.py

```diff
@@ -40,6 +40,10 @@ def setting_function_good(request):
     "(",                                                                # Syntax error.
     "[x for x in (1).__class__.__base__.__subclasses__() if x.__name__ == '_ImportLockContext'][0]().__enter__.__globals__['__builtins__']['__import__']('os').system(\"echo omgzomg\")",  # Obsfucated call to system
     "[x for x in (1).__class__.__base__.__subclasses__() if x.__name__ == '_ImportLockContext'][0]().__enter__.__globals__['__' + 'builtins__']['__import__']('os').system(\"echo omgzomg\")"   #Another obsfucated call to system
+    "'_'", # This string is not allowed
+    "import sys",  # We don't allow importing
+    "'a_a'[1]",  # Trying to circumvent the protection against underscores
+    "{x:x for x in [1,2,3,4]}"  # We don't allow dict comprehension
```

Comment:
Why don't we allow dict comprehension? Seems arbitrary to disallow that.

Reply:
One of the hacks from the pen testing guys used list comprehensions to get past our defences. You can use the same trick with dict comprehensions. The main reason for it is that you can use it to filter results by using the "if" option of a comprehension.